horace_core/sqw/PixelData/@PixelData/PixelData.m

```diff
@@ -427,27 +430,33 @@ function delete(obj)
                 has_more = obj.pix_position_ + obj.base_page_size  <= obj.num_pixels;
             end
         end
-        
-        function [current_page_num,total_num_pages] = advance(obj)
+
+        function [current_page_num, total_num_pages] = advance(obj, varargin)
             % Load the next page of pixel data from the file backing the object
             %
             % This function will throw a PIXELDATA:advance error if attempting to
             % advance past the final page of data in the file.
             %
             % This function does nothing if the pixel data is not file-backed.
             %
-            % Returns:
-            % current page number and total number of pages advance will
+            %  >> obj.advance()
+            %  >> obj.advance('nosave', true)
+            %
+            % Inputs:
+            % -------
+            % nosave  Keyword argument. Set to true to discard changes to cache.
+            %         (default: false)
+            %
+            % Outputs:
+            % --------
+            % current_page_number  The new page and total number of pages advance will
             % walk through to complete the algorithm
             %
             current_page_num = 1;
-            total_num_pages  = 1;
+            total_num_pages = 1;
```

Comment:
`page_is_dirty() / write_dirty_page_()` has been dropped-- where is that logic now to ensure dirty pages are written on a call to `advance()`?

Reply:
It's in `move_to_page`